src/telemetry/Manager.h

```diff
@@ -2,161 +2,62 @@
-         * Helper struct representing a histogram as sum and buckets.
-         * @tparam T The data type used by the histogram (double or int64_t).
-         */
-        template<class T>
-        struct HistogramData {
-            T sum;
-            std::vector<Bucket<T>> buckets;
-        };
-
-        using DblHistogramData = HistogramData<double>;
-        using IntHistogramData = HistogramData<int64_t>;
-
-        /**
-         * Constructor.
-         * @param family Broker layer family handle for this metric.
-         * @param label_values The string values for each of the metric's labels.
-         * @param histogram The histogram's data (sum and individual buckets).
-         */
-        CollectedHistogramMetric(const broker::telemetry::metric_family_hdl* family,
-                                 std::vector<std::string_view> label_values,
-                                 std::variant<DblHistogramData, IntHistogramData> histogram)
-
-            : family(family), label_values(std::move(label_values)), histogram(std::move(histogram)) {}
-
-        const broker::telemetry::metric_family_hdl* family;
-        std::vector<std::string_view> label_values;
-        std::variant<DblHistogramData, IntHistogramData> histogram;
-
-        /**
-         * @return A script layer Telemetry::HistogramMetric record for this histogram.
-         */
-        zeek::RecordValPtr AsHistogramMetricRecord() const;
-    };
-
-    /**
-     * @return A script layer Telemetry::MetricOpts record for the given metric family.
-     * @param metric_typ The type of metric.
-     * @param family Broker layer family handle for the family.
-     * @tparam T The underlying data type (double or int64_t)
-     */
-    template<typename T>
-    zeek::RecordValPtr GetMetricOptsRecord(MetricType metric_type, const broker::telemetry::metric_family_hdl* family);
-
-    /**
-     * @return All counter and gauge metrics and their values matching prefix and name.
+     * @return A VectorVal containing all counter and gauge metrics and their values matching prefix and name.
      * @param prefix The prefix pattern to use for filtering. Supports globbing.
      * @param name The name pattern to use for filtering. Supports globbing.
      */
-    std::vector<CollectedValueMetric> CollectMetrics(std::string_view prefix, std::string_view name);
+    ValPtr CollectMetrics(std::string_view prefix, std::string_view name);
 
     /**
-     * @return All histogram metrics and their data matching prefix and name.
+     * @return A VectorVal containing all histogram metrics and their values matching prefix and name.
      * @param prefix The prefix pattern to use for filtering. Supports globbing.
      * @param name The name pattern to use for filtering. Supports globbing.
      */
-    std::vector<CollectedHistogramMetric> CollectHistogramMetrics(std::string_view prefix, std::string_view name);
+    ValPtr CollectHistogramMetrics(std::string_view prefix, std::string_view name);
```

Comment:
Just curious: is there a reason here to have separate APIs for metrics/gauges vs histograms? Just convenience?

Reply:
The record types returned are different as well.